labs/b4.md

```diff
@@ -0,0 +1,165 @@
+---
+title: Beginner Lab 4 - Debian, packages, compiling software
+layout: lab
+---
+
+## What is a distribution?
+
+A Linux distribution is a combination of the Linux kernel, package management system, drivers, scheduler, and other software. They are the operating systems that use Linux as a kernel. 
+
+## What should I use then?
+
+Currently, there are around 300 actively developed distributions listed on DistroWatch. It might be a bit overwhelming with so many choices. Major distributions include Debian, Arch Linux, and Red Hat Linux, but in this course, we’ll only be using Debian.
+
+### What is Debian?
+
+Debian is a distribution developed by the Debian Project beginning in 1993. This distribution strives to provide a completely free (libre) software experience. Many other distributions are based on Debian as Debian releases are usually very stable. One such distribution is Ubuntu, which is very popular and has spawned its many of its own derivatives. What unifies all of Debian-based distributions is the usage of the DEB package format and the dpkg package manager.
+
+## But what about these “package managers”?
+
+If you come from the Windows or even MacOS ecosystem, you might be scratching your head thinking what a package managers is. On those operating systems, the most conventional and used method of installing programs is to run an installer that copies the files from an archive to the correct place (and makes changes to a registry of some sorts).
```

Comment:
Grammar fix: `thinking what a package managers is` -> `thinking about what a package manager is` (or something similar)